app/models/event.rb

```diff
@@ -9,4 +9,11 @@ def start_time_must_be_lower_than_end_time
     errors.add(:start_time, "Can't be higher than end_time") if start_time.try(:>, end_time)
   end
 
+  scope :in_week, ->(week) {
+    query = '(start_time >= :start_week AND start_time <= :end_week)'\
+    'OR (end_time >= :start_week AND end_time <= :end_week )'\
+    'OR (start_time <= :start_week AND end_time >= :end_week)'
+    where(query, start_week: week.beginning_of_week, end_week: week.end_of_week)
```

Comment:
I think you can simplify this condition to:

`:end_week >= start_time AND :start_week <= end_time`